Fix endpoint-related names to use consistent singular/plural

danwinship opened the pull request:
#### What type of PR is this?
/kind cleanup
/sig network

#### What this PR does / why we need it:
See below, https://github.com/kubernetes/kubernetes/pull/118729#issuecomment-1753915385

~Due to the fact that `v1.Endpoints` has a plural name, lots of endpoint-related things in kube-proxy used "`endpoints`" in their names even when they were referring to a single object, while others used just "`endpoint`", generally in reference to the more generic concept. This always made it hard to remember the names correctly, and that went from "understandably confusing" to "pointlessly confusing" when we replaced all usage of `v1.Endpoints` with `discoveryv1.EndpointSlice`.~

~Fix the names of endpoint-related types/functions/variables to use singular "`Endpoint`" in all cases where the corresponding service-related types/functions/variables use singular "`Service`"/"`ServicePort`".~

~(Also fix some comments that referred to other things by the wrong
names...)~

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

danwinship commented:
/assign @aojea

aojea commented:
this LGTM is just renaming, defer to @thockin for approval

/assign @thockin

thockin commented on pkg/proxy/endpoints.go:
```diff
@@ -282,13 +282,13 @@ func getLastChangeTriggerTime(annotations map[string]string) time.Time {
 }
 
-// endpointsChange contains all changes to endpoints that happened since proxy
+// endpointChange contains all changes to endpoints that happened since proxy
```
Should this be "endpointChanges" - is *is* plural, and the new name makes it sound singular.

danwinship replied:
"change" rather than "changes" is correct IMO; it describes how the endpoint IPs for a particular service have changed since the last sync. Regardless of how many events there were and how many EndpointSlice objects are involved and how many endpoint IPs they have, the object here is only describing a single change: the change between what we synced last time, and what we should sync next time.

Arguably it could/should stay as "endpoints", not "endpoint", because it encompasses all of the endpoint IPs associated with a particular service...

thockin replied:
Yeah, SOMETHING is plural here - I don't much care which half, but `endpointChanges` makes most sense to me

thockin commented on pkg/proxy/iptables/proxier.go:
```diff
@@ -788,13 +788,13 @@ func (proxier *Proxier) syncProxyRules() {
 	}()
 
-	var serviceChanged, endpointsChanged sets.Set[string]
+	var serviceChanged, endpointChanged sets.Set[string]
```
this is a place where plural makes sense to me - the set of endpoints changed

thockin replied:
also `servicesChanged` makes sense to me :)

thockin reviewed (commented):
Most of this seems good or neutral.  A few feel worse now.

This would be much easier to review if it was one commit per change :)

danwinship commented:
/hold

> Fix the names of endpoint-related types/functions/variables to use singular "`Endpoint`" in all cases where the corresponding service-related types/functions/variables use singular "`Service`"/"`ServicePort`".

That may be wrong since the service objects always only refer to a single service, while the endpoint(s) objects generally refer to multiple endpoint IPs even if they refer to only a single EndpointSlice.

danwinship committed:
Fix "Endpoint" vs "Endpoints" in proxy type names

The use of "Endpoint" vs "Endpoints" in these type names is tricky
because it doesn't always make sense to use the same singular/plural
convention as the corresonding service-related types, since often the
service-related type is referring to a single service while the
endpoint-related type is referring to multiple endpoint IPs.

The "endpointsInfo" types in the iptables and winkernel proxiers are
now "endpointInfo" because they describe a single endpoint IP (and
wrap proxy.BaseEndpointInfo).

"UpdateEndpointMapResult" is now "UpdateEndpointsMapResult", because
it is the result of EndpointsMap.Update (and it's clearly correct for
EndpointsMap to have plural "Endpoints" because it's a map to an array
of proxy.Endpoint objects.)

"EndpointChangeTracker" is now "EndpointsChangeTracker" because it
tracks changes to the full set of endpoints for a particular service
(and the new name matches the existing "endpointsChange" type and
"Proxier.endpointsChanges" fields.)

danwinship commented:
/retitle Fix endpoint-related names to use consistent singular/plural
/hold cancel

thockin reviewed (commented):
Thanks!

/lgtm
/approve

danwinship commented:
/retest-required
